### PR TITLE
refactor: Refactored Odp Code to get rid of platform specific checks

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
Refactored Odp Code to get rid of platform specific checks.

## Test plan
Not tested at all. Relying on John to test and fix unit tests if needed

## Issues

Merging into changes related to [FSSDK-9023](https://jira.sso.episerver.net/browse/FSSDK-9023)

### CODE REVIEW

File: packages/optimizely-sdk/lib/plugins/odp_manager/index.node.ts [hunk 1]
```diff
@@ -55,5 +57,32 @@ export class NodeOdpManager extends OdpManager {
...
     });
+
+    // Set up Events Manager (Events REST API Interface)
+    if (odpOptions?.eventManager) {
+      this.eventManager = odpOptions.eventManager;
+      this.eventManager.updateSettings(this.odpConfig);
+    } else {
+      this.eventManager = new NodeOdpEventManager({
+        odpConfig: this.odpConfig,
+        apiManager: new NodeOdpEventApiManager(nodeRequestHandler, this.logger),
```
Comment (jaeopt) [inline packages/optimizely-sdk/lib/plugins/odp_manager/index.node.ts:68]: override with `customRequestHandler` if given.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks great! I see some platform dependent logic for segmentManager as well. If we can have consistent structure for both segmentManager and eventManager, it may look better.

**Comment by opti-jnguyen** [comment]
@jaeopt - went ahead and resolved tests and finished splitting off odpmanager responsibilities into variant-specific implementations.

**Review by jaeopt — APPROVED** [review]
LGTM

File: packages/optimizely-sdk/lib/index.node.ts [hunk 2]
```diff
@@ -100,9 +94,9 @@ const createInstance = function(config: Config): Client | null {
       dispatcher: config.eventDispatcher || defaultEventDispatcher,
       flushInterval: eventFlushInterval,
       batchSize: eventBatchSize,
-      maxQueueSize:  config.eventMaxQueueSize || DEFAULT_EVENT_MAX_QUEUE_SIZE,
+      maxQueueSize: config.eventMaxQueueSize || DEFAULT_EVENT_MAX_QUEUE_SIZE,
       notificationCenter,
-    }
```
[the remark this answers is not in the document]
Comment (jaeopt) [reply packages/optimizely-sdk/lib/index.node.ts:105]: I see this is for logx events only. No need to change :)

**Pull request merged**